src/psyclone/psyir/symbols/containersymbol.py

```diff
@@ -125,15 +125,28 @@ def copy(self):
         new_symbol.is_intrinsic = self.is_intrinsic
         return new_symbol
 
-    @property
-    def container(self):
+    def container(self, local_node=None):
```

Comment:
Since it's not a property anymore I would name it "get_container_psyir" to be equivalent to the introduced "get_routine_psyir".

And in my opinion, in both cases, "search" (instead of "get") would imply more clearly that they are potentially expensive operation that can return "None". But I will accept both.

Reply:
That's a good idea. I've gone with `find_{container,routine}_psyir` and I think it makes things a lot more readable.